internal/pkg/errorconcept/handler.go

```diff
@@ -0,0 +1,68 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+package errorconcept
+
+import (
+	"net/http"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+)
+
+type ErrorConceptType interface {
+	httpErrorCode() int
+	isA(err error) bool
+	message(err error) string
+}
+
+type handler struct {
+	logger logger.LoggingClient
+}
+
+type ErrorHandler interface {
+	HandleAll(w http.ResponseWriter, err error, allowableErrors []ErrorConceptType, defaultError ErrorConceptType)
+	HandleOne(w http.ResponseWriter, err error, allowableError ErrorConceptType, defaultError ErrorConceptType)
+}
+
+func NewErrorHandler(l logger.LoggingClient) ErrorHandler {
+	h := handler{l}
+	return &h
+}
+
+func (e *handler) handle(w http.ResponseWriter, err error, ec ErrorConceptType) {
```

Comment:
Look in previous comments for suggested implementation. 

handle -> Handle; modify to deal with a single error.  

Interface segregation principle -- HandleOne is doing double duty.  Call public Handle() to deal with a specific error instead of calling HandleOne() with nil for allowableError.